The job `x86_64-gnu-llvm-7` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=15587&view=logs&jobId=0230d20d-e0df-5b62-a6ec-fad0f9c3c3d5) ([pretty log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/15587/logs/65), [raw log](https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/15587/logs/65)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-12-10T07:21:44.0024804Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-12-10T07:21:44.0218503Z ##[command]git config gc.auto 0
2019-12-10T07:21:44.0304536Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-12-10T07:21:44.0357904Z ##[command]git config --get-all http.proxy
2019-12-10T07:21:44.0511355Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/62514/merge:refs/remotes/pull/62514/merge
---
2019-12-10T07:27:54.1035366Z Done!
2019-12-10T07:27:54.1035499Z some tidy checks failed
2019-12-10T07:27:54.1036079Z 
2019-12-10T07:27:54.1036110Z 
2019-12-10T07:27:54.1037076Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/checkout/src" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "--no-vendor"
2019-12-10T07:27:54.1037506Z 
2019-12-10T07:27:54.1037533Z 
2019-12-10T07:27:54.1045060Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/tools/tidy
2019-12-10T07:27:54.1045416Z Build completed unsuccessfully in 0:01:31
2019-12-10T07:27:54.1045416Z Build completed unsuccessfully in 0:01:31
2019-12-10T07:27:54.1107593Z == clock drift check ==
2019-12-10T07:27:54.1117175Z   local time: Tue Dec 10 07:27:54 UTC 2019
2019-12-10T07:27:54.2626516Z   network time: Tue, 10 Dec 2019 07:27:54 GMT
2019-12-10T07:27:54.2632496Z == end clock drift check ==
2019-12-10T07:27:55.5930874Z 
2019-12-10T07:27:55.6034533Z ##[error]Bash exited with code '1'.
2019-12-10T07:27:55.6065206Z ##[section]Starting: Checkout
2019-12-10T07:27:55.6067754Z ==============================================================================
2019-12-10T07:27:55.6067831Z Task         : Get sources
2019-12-10T07:27:55.6067897Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

--- a/src/liballoc/boxed.rs
+++ b/src/liballoc/boxed.rs
@@ -65,7 +65,7 @@
 //!
 //! So long as `T: Sized`, a `Box<T>` is guaranteed to be represented as a
 //! single pointer and is also ABI-compatible with C pointers (i.e. the C type
-//! `T*`). This means that you have Rust code which passes ownership of a
+//! `T*`). This means that you can have Rust code which passes ownership of a
 //! `Box<T>` to C code by using `Box<T>` as the type on the Rust side, and
 //! `T*` as the corresponding type on the C side. As an example, consider this
 //! C header which declares functions that create and destroy some kind of

--- a/src/liballoc/boxed.rs
+++ b/src/liballoc/boxed.rs
@@ -73,14 +73,14 @@
 //!
 //! ```c
 //! /* C header */
-//! 
+//!
 //! /* Returns ownership to the caller */
 //! struct Foo* foo_new(void);
-//! 
+//!
 //! /* Takes ownership from the caller; no-op when invoked with NULL */
 //! void foo_delete(struct Foo*);
 //! ```
-//! 
+//!
 //! These two functions might be implemented in Rust as follows. Here, the
 //! `struct Foo*` type from C is translated to `Box<Foo>`, which captures
 //! the ownership constraints. Note also that the nullable argument to
@@ -99,7 +99,7 @@
 //! #[no_mangle]
 //! pub extern "C" fn foo_delete(_: Option<Box<Foo>>) {}
 //! ```
-//! 
+//!
 //! Even though `Box<T>` has the same representation and C ABI as a C pointer,
 //! this does not mean that you can convert an arbitrary `T*` into a `Box<T>`
 //! and expect things to work. `Box<T>` values will always be fully aligned,

--- a/src/liballoc/boxed.rs
+++ b/src/liballoc/boxed.rs
@@ -63,13 +63,14 @@
 //! T` obtained from `Box::<T>::into_raw` may be deallocated using the
 //! [`Global`] allocator with `Layout::for_value(&*value)`.
 //!
-//! So long as `T: Sized`, a `Box<T>` is guaranteed to be represented as a
-//! single pointer and is also ABI-compatible with C pointers (i.e. the C type
-//! `T*`). This means that you can have Rust code which passes ownership of a
-//! `Box<T>` to C code by using `Box<T>` as the type on the Rust side, and
-//! `T*` as the corresponding type on the C side. As an example, consider this
-//! C header which declares functions that create and destroy some kind of
-//! `Foo` value:
+//! So long as `T: Sized`, a `Box<T>` is guaranteed to be represented
+//! as a single pointer and is also ABI-compatible with C pointers
+//! (i.e. the C type `T*`). This means that if you have extern "C"
+//! Rust functions that will be called from C, you can define those
+//! Rust functions using `Box<T>` types, and use `T*` as corresponding
+//! type on the C side. As an example, consider this C header which
+//! declares functions that create and destroy some kind of `Foo`
+//! value:
 //!
 //! ```c
 //! /* C header */
@@ -108,6 +109,14 @@
 //! is to only use `Box<T>` for pointers that originated from the global
 //! allocator.
 //!
+//! **Important.** At least at present, you should avoid using
+//! `Box<T>` types for functions that are defined in C but invoked
+//! from Rust. In those cases, you should directly mirror the C types
+//! as closely as possible. Using types like `Box<T>` where the C
+//! definition is just using `T*` can lead to undefined behavior, as
+//! described in [rust-lang/unsafe-code-guidelines#198][ucg#198].
+//!
+//! [ucg#198]: https://github.com/rust-lang/unsafe-code-guidelines/issues/198
 //! [dereferencing]: ../../std/ops/trait.Deref.html
 //! [`Box`]: struct.Box.html
 //! [`Global`]: ../alloc/struct.Global.html

